sparkle/platform/output.py

```diff
@@ -0,0 +1,298 @@
+
+        cutoff_time = self.configurator.scenario.cutoff_time
+        penalty = penalty_multiplier * \
+            self.configurator.scenario.cutoff_time
+        perf_par_conf = sgrfch.get_par_performance(val_default,
+                                                   cutoff_time,
+                                                   penalty,
+                                                   objective)
+        perf_par_def = sgrfch.get_par_performance(val_conf,
+                                                  cutoff_time,
+                                                  penalty,
+                                                  objective)
+        performance = ConfigurationPerformance(perf_par_conf, perf_par_def)
+
+        results = ConfigurationResults(performance, results_default, results_conf)
+        return results, best_config
+
+    def write_output(self: ConfigurationOutput) -> None:
+        """Write data into a json file."""
+        def serialize_configuration_results(cr: ConfigurationResults) -> None:
+            return {
+                "performance": {
+                    "configured_PAR10": cr.performance.configured_PAR10,
+                    "default_PAR10": cr.performance.default_PAR10,
+                },
+                "configured_results": {
+                    "solver": cr.configured_results.solver,
+                    "configuration": cr.configured_results.configuration,
+                    "instance_set": cr.configured_results.instance_set,
+                    "result_header": cr.configured_results.result_header,
+                    "result_vals": cr.configured_results.result_vals,
+                },
+                "default_results": {
+                    "solver": cr.default_results.solver,
+                    "configuration": cr.default_results.configuration,
+                    "instance_set": cr.default_results.instance_set,
+                    "result_header": cr.default_results.result_header,
+                    "result_vals": cr.default_results.result_vals,
+                }
+            }
+
+        output_data = {
+            "solver": self.solver.name if self.solver else None,
+            "configurator": (
+                str(self.configurator.executable_path) if self.configurator else None
+            ),
+            "best_config": self.best_config,
+            "training_set": (
+                serialize_configuration_results(self.training) if self.training else None
+            ),
+            "test_set": (
+                serialize_configuration_results(self.test) if self.test else None
+            ),
+        }
+
+        Path(os.path.dirname(self.output)).mkdir(exist_ok=True)
+        with self.output.open("w") as f:
+            json.dump(output_data, f, indent=4)
+        print("Analysis of configuration can be found here: ", self.output)
+
+
+if __name__ == "__main__":
+    global settings
+    gv.settings = Settings()
```

Comment:
gv.settings is a method now instead of a property in the latest dev version

Reply:
Please fix

Reply:
This isn't a real part of the class
This is just for testing it at the moment
In the future, when the code is accessed from generate_report (or somewhere else), we just call the class and don't actually run the file (at least that was my understanding so far)